apps/app.go

```diff
@@ -163,13 +168,28 @@ type indexedPortDefinition struct {
 
 func (app App) findConsulPortDefinitions() []indexedPortDefinition {
 	var definitions []indexedPortDefinition
-	for i, d := range app.PortDefinitions {
+	for i, d := range extractPortDefinitions(&app) {
 		if _, ok := d.Labels[MarathonConsulLabel]; ok {
 			definitions = append(definitions, indexedPortDefinition{
 				Index:  i,
 				Labels: d.Labels,
 			})
 		}
 	}
+
 	return definitions
 }
+
+// Deprecated: Allows for backward compatibility with Marathons' network API
+// PortDefinitions are deprecated in favor of Marathons' new PortMappings
+// see https://github.com/mesosphere/marathon/pull/5391
+func extractPortDefinitions(app *App) []PortDefinition {
```

Comment:
Why not a struct receiver? 

`func (app App) extractPortDefinitions()`

Reply:
Good suggestion. I'll update.